src/components/tooltip/tooltip.component.js

```diff
@@ -1,70 +1,97 @@
+    {
+      children,
+      isVisible,
+      position = "top",
+      message,
+      type,
+      size = "M",
+      isPartOfInput,
+      inputSize,
+      id,
+      ...props
+    },
+    ref
+  ) => {
+    const tooltipRef = useRef(ref || null);
 
-  static defaultProps = {
-    align: "center",
-    position: "top",
-    isVisible: false,
-  };
+    const tooltip = (attrs, content) => {
+      const currentPosition = attrs["data-placement"] || position;
+      const positionArrowAbove = ["bottom", "right"].includes(currentPosition);
 
-  get tooltipHTML() {
-    const { children, onMouseEnter, onMouseLeave, ...commonProps } = this.props;
+      const pointer = (
+        <StyledPointer
+          key="pointer"
+          type={type}
+          {...attrs}
+          position={currentPosition}
+          data-popper-arrow=""
+          data-element="tooltip-pointer"
+        />
+      );
+
+      return (
+        <StyledTooltip
+          data-element="tooltip"
+          role="tooltip"
+          tabIndex="-1"
+          type={type}
+          size={size}
+          id={id}
+          {...tagComponent("tooltip", props)}
+          isPartOfInput={isPartOfInput}
+          inputSize={inputSize}
+          {...attrs}
+          position={currentPosition}
+          ref={tooltipRef}
+        >
+          {positionArrowAbove && pointer}
+          <div>{content}</div>
+          {!positionArrowAbove && pointer}
+        </StyledTooltip>
+      );
+    };
 
     return (
-      <StyledTooltipWrapper
-        role="tooltip"
-        onMouseEnter={onMouseEnter}
-        onMouseLeave={onMouseLeave}
-        {...commonProps}
-        {...tagComponent("tooltip", this.props)}
+      <Tippy
+        placement={position}
+        delay={100}
+        {...(isVisible !== undefined && { visible: isVisible })}
+        render={(attrs, content) => tooltip(attrs, message || content)}
       >
-        <StyledTooltipInner {...commonProps}>
-          <>
-            {children}
-            <StyledTooltipPointer
-              key="pointer"
-              {...commonProps}
-              data-element="tooltip-pointer"
-            />
-          </>
-        </StyledTooltipInner>
-      </StyledTooltipWrapper>
+        {children}
+      </Tippy>
     );
   }
+);
 
-  render() {
-    return this.props.isVisible && this.props.children
-      ? this.tooltipHTML
-      : null;
-  }
-}
+Tooltip.propTypes = {
+  /** The message to be displayed within the tooltip */
+  message: PropTypes.node.isRequired,
+  /** The id attribute to use for the tooltip */
+  id: PropTypes.string,
+  /** Whether to to show the Tooltip */
+  isVisible: PropTypes.bool,
+  /** Sets position of the tooltip */
+  position: PropTypes.oneOf(["top", "bottom", "left", "right"]),
+  /** Defines the message type */
+  type: PropTypes.string,
+  /** Children elements */
+  children: PropTypes.node.isRequired,
+  /** Defines the size of the tooltip content */
+  size: PropTypes.oneOf(["M", "L"]),
```

Comment:
Maybe this should be `default` and `L` (or `large`) - `M` doesn't really mean anything when there's no small

Reply:
Yeah could do, I chose the naming convention described in the DLS and I know we've used the short form for sizes in other components recently. I'll check with UX as if there's not going to be other size variants then it makes sense to simplify the interface the way you suggest, or at least maybe go with `medium` and `large`